src/lib.rs

```diff
@@ -173,164 +194,388 @@ impl<const OSC_RAW_BUF_SIZE: usize> Parser<OSC_RAW_BUF_SIZE> {
+                self.state = State::DcsIntermediate
+            },
+            0x30..=0x39 => self.action_paramnext(byte),
+            0x3A => self.action_subparam(),
+            0x3B => self.action_param(),
+            0x3C..=0x3F => self.state = State::DcsIgnore,
+            0x40..=0x7E => self.action_hook(performer, byte),
+            0x7F => self.state = State::DcsIgnore,
+            _ => self.anywhere(performer, byte),
+        }
+    }
+
+    #[inline(always)]
+    fn advance_dcs_passthrough<P: Perform>(&mut self, performer: &mut P, byte: u8) {
+        match byte {
+            0x00..=0x17 | 0x19 | 0x1C..=0x7E => performer.put(byte),
+            0x7F => (),
+            0x18 | 0x1A => {
+                performer.unhook();
+                performer.execute(byte);
+                self.state = State::Ground
+            },
+            0x1B => {
+                performer.unhook();
+                self.reset_params();
+                self.state = State::Escape
+            },
+            0x9C => {
+                performer.unhook();
+                self.state = State::Ground
+            },
+            _ => (),
+        }
+    }
+
+    #[inline(always)]
+    fn advance_esc<P: Perform>(&mut self, performer: &mut P, byte: u8) {
+        match byte {
+            0x00..=0x17 | 0x19 | 0x1C..=0x1F => performer.execute(byte),
+            0x20..=0x2F => {
+                self.action_collect(byte);
+                self.state = State::EscapeIntermediate
+            },
+            0x30..=0x4F => {
+                performer.esc_dispatch(self.intermediates(), self.ignoring, byte);
+                self.state = State::Ground
+            },
+            0x50 => {
+                self.reset_params();
+                self.state = State::DcsEntry
+            },
+            0x51..=0x57 => {
+                performer.esc_dispatch(self.intermediates(), self.ignoring, byte);
+                self.state = State::Ground
+            },
+            0x58 => self.state = State::SosPmApcString,
+            0x59..=0x5A => {
+                performer.esc_dispatch(self.intermediates(), self.ignoring, byte);
+                self.state = State::Ground
+            },
+            0x5B => {
+                self.reset_params();
+                self.state = State::CsiEntry
+            },
+            0x5C => {
+                performer.esc_dispatch(self.intermediates(), self.ignoring, byte);
+                self.state = State::Ground
+            },
+            0x5D => {
+                self.osc_raw.clear();
+                self.osc_num_params = 0;
+                self.state = State::OscString
+            },
+            0x5E..=0x5F => self.state = State::SosPmApcString,
+            0x60..=0x7E => {
+                performer.esc_dispatch(self.intermediates(), self.ignoring, byte);
+                self.state = State::Ground
+            },
+            _ => self.anywhere(performer, byte),
+        }
+    }
```

Comment:
Generally I appreciate having everything laid out in byte order. But honestly this function would be half as long if we didn't do it.

Reply:
they are in byte order though here.